src/acrcssc/azext_acrcssc/helper/_taskoperations.py

```diff
@@ -145,22 +145,39 @@ def _eval_trigger_run(cmd, registry, resource_group, run_immediately):
         _trigger_task_run(cmd, registry, resource_group, CONTINUOUSPATCH_TASK_SCANREGISTRY_NAME)
 
 
-def delete_continuous_patch_v1(cmd, registry, dryrun):
+def delete_continuous_patch_v1(cmd, registry, yes):
     logger.debug("Entering delete_continuous_patch_v1")
     cssc_tasks_exists, _ = check_continuous_task_exists(cmd, registry)
     cssc_config_exists = check_continuous_task_config_exists(cmd, registry)
-    if not dryrun and (cssc_tasks_exists or cssc_config_exists):
+
+    acr_run_client = cf_acr_runs(cmd.cli_ctx)
+    resource_group_name = parse_resource_id(registry.id)[RESOURCE_GROUP]
+    running_tasks = WorkflowTaskStatus.get_taskruns_with_filter(
+        acr_run_client,
+        registry_name=registry.name,
+        resource_group_name=resource_group_name,
+        status_filter=[TaskRunStatus.Running.value, TaskRunStatus.Queued.value, TaskRunStatus.Started.value],
+        taskname_filter=[CONTINUOUSPATCH_TASK_SCANREGISTRY_NAME, CONTINUOUSPATCH_TASK_SCANIMAGE_NAME, CONTINUOUSPATCH_TASK_PATCHIMAGE_NAME])
+    if len(running_tasks) > 0:
+        from knack.prompting import prompt_y_n
+        if yes or prompt_y_n("There are currently running tasks for this workflow. Do you want to cancel their execution?"):
```

Comment:
Suggestion: log which tasks they are about to cancel

Reply:
adding debug message to `_cancel_task_runs()`